include/flatbuffers/flatbuffers.h

```diff
@@ -576,12 +576,14 @@ class vector_downward {
   #ifndef FLATBUFFERS_CPP98_STL
   // Relinquish the pointer to the caller.
   unique_ptr_t release() {
-    // Actually deallocate from the start of the allocated memory.
-    std::function<void(uint8_t *)> deleter(
-      std::bind(&simple_allocator::deallocate, allocator_, buf_));
+    // Save some vars to local scope for capture below
+    const simple_allocator& allocator = allocator_;
+    uint8_t* oldbuf = buf_;
 
-    // Point to the desired offset.
-    unique_ptr_t retval(data(), deleter);
+    // Point to the desired offset, but delete from the start of alloc mem
+    unique_ptr_t retval(data(), [&allocator, oldbuf](uint8_t *) {
```

Comment:
Yep, the ownership is a little incorrect in this PR, actually. Turns out the previous deleter was copying the allocator via copy ctor (notice it was binding to `allocator_` and not `&allocator_`), so the old code was safe. `SliceAllocator` in this PR is not copy-constructible because it's stateful, so the old code wasn't working and I slapped the lambda on it, but that's not really right either.

The other PR I just sent #4312 fixes this for both scenarios. Once that's addressed I'll rebase this PR.